### ISSUE TITLE
Error: No such command "depoly".

### ISSUE THREAD

**Issue by tuenrobot** [miguelgrinberg/flasky#435]
In Chapter 17, testing by heroku local,

I am having trouble running this command
`heroku local:run flask deploy
`

I get the following output
```
[OKAY] Loaded ENV .env File as KEY=VALUE Format
Usage: flask [OPTIONS] COMMAND [ARGS]...

Error: No such command "deploy".
```
while it should be
```
[OKAY] Loaded ENV .env File as KEY = VALUE Format 
INFO Context impl SQLiteImpl. INFO Will assume non-transactional DDL. 
INFO Running upgrade -> 38c4e85512a9, initial migration
...
```

I have no idea why there is no such command.
Could you give me any suggestions, please?
Thank you.

**Comment by miguelgrinberg** [comment]
Do you have the deploy command in your code? This is what I mean: https://github.com/miguelgrinberg/flasky/blob/master/flasky.py#L73.

If you do, did you set the `FLASK_APP` environment variable appropriately?

**Comment by tuenrobot** [comment]
I somehow missed the deploy command in my code.
Thank you for pointing where the command is!

**Comment by joecerniglia** [comment]
Hi Miguel,
Thanks for this interesting book!

I've been having some trouble with Heroku deployment. I needed to do some alterations to the versions in the requirements files to allow the git push to work. Then I worked with the app on a branch, as you instructed here:
https://github.com/miguelgrinberg/flasky/issues/524
My alterations to common.txt in requirements, done by repeated trial and error, are here:
alembic==0.9.3
bleach==2.0.0
blinker==1.4
click==7.1.2
dominate==2.3.1
Flask==1.1.4
Flask-Bootstrap==3.3.7.1
Flask-HTTPAuth==3.2.3
Flask-Login==0.4.0
Flask-Mail==0.9.1
Flask-Migrate==2.0.4
Flask-Moment==0.5.1
Flask-PageDown==0.2.2
Flask-SQLAlchemy==2.2
Flask-WTF==0.14.2
html5lib==0.999999999
itsdangerous
Jinja2==2.11.3
Mako==1.0.7
Markdown==2.6.8
MarkupSafe==2.0.1
pyparsing==2.4.7
python-dateutil==2.6.1
python-editor==1.0.3
setuptools==57.5.0
six==1.10.0
SQLAlchemy==1.4.37
visitor==0.1.3
webencodings==0.5.1
Werkzeug==1.0.1
WTForms==2.1

My alterations to heroku.txt in requirements are here:
-r prod.txt
Flask-SSLify==0.1.5
gunicorn==19.7.1
psycopg2-binary

A summary of these changes to both files is here:
To common.txt: 
Werkzeug==1.0.1
MarkupSafe==2.0.1
Jinja2==2.11.3
Flask==1.1.4
Itsdangerous
click==7.1.2
sqlalchemy==1.4.37

Add:
pyparsing==2.4.7

Add:
setuptools==57.5.0

========
To heroku.txt: psycopg2-binary

These changes allowed me to do the git push. The commands I ran in sequence are:
git reset --hard
#manually make changes to 2 requirements files
git checkout 17c
git branch -D my17c #delete previous attempts at making a branch on Heroku server
git checkout -b my17c #checkout to the branch the local revisions to requirements
git add requirements/heroku.txt   
git add requirements/common.txt   
git commit -m "switch to psycopg2-binary and switch to Werkzeug==0.14.1"
git push -f heroku master my17c:main 

Then, however, when I try to run "heroku run flask deploy",  I receive the error "No such command 'deploy'.
I have my deploy code in flasky.py, and I have set my FLASK_APP variable to be flasky.py.  I've set up my .env and my Procfile as indicated.

I have tried to resolve this on my own in a number of ways but now I appear to have reached an impasse.  

Can you help, Miguel?

I can include the text from my console in a separate reply if that is of help.  I've saved the work.

Thanks in advance.  I am enjoying your book and working hard on it.

Sincerely,
Joe Cerniglia

**Comment by joecerniglia** [comment]
I now have an intermediate version of the Flasky app up and running on Heroku. I studied a video on deployment, https://www.youtube.com/watch?v=FKy21FnjKS0&list=LLqN8hAEvNxLmWob8su8oQBg, which provided a perspective on deployment that makes use of the Heroku website in tandem with the CLI.  First, I deployed the videographer, Anthony's, website app, "Ask a question," and then I set to work again on Flasky. I note that the DATABASE_URL must now have "postgresql:// and not "postgres://". I learn as much by practice as anything else, and not giving up. I plan now to return to the earlier chapters and try to deploy successive git checkouts. I have a flow of work and a toolbox in place that I think can help me overcome challenges. One thing is for sure and that is that the tools and the syntax behind them are not remaining constant. Rather, they seem to be evolving quite rapidly, especially in the field of web design.  Best, Joe
